### PR TITLE
Fix 1mom optics for RRTMG

### PR DESCRIPTION
**Pull request by brhillman**
Fix use of cloud optics for sam1mom microphysics when using RRTMG. Previously, we hard-coded the `oldcldoptics` flag when using sam1mom microphysics to try to force RRTMG to use the "old" cloud optics schemes, which do not use information from MG2 to specify particle size distribution parameters. However, this flag appears to have inconsistent behavior between the shortwave and longwave codes, and in the longwave it was just being used to add an extra diagnostic call to the old optics routines, before just being overwritten with calls to the newer optics routines anyways. Here, we fix the behavior by explicitly setting the liquid and ice optics schemes for sam1mom via edits to `build-namelist` to make sure the defaults for sam1mom are the older "Slingo" and "Ebert-Curry" optics. This will be non-BFB for tests that used sam1mom microphysics with RRTMG, but should be BFB for all RRTMGP tests, and for m2005 microphysics tests.

### CODE REVIEW

File: components/cam/src/physics/cam/rad_constituents.F90 [hunk 1]
```diff
@@ -297,17 +295,6 @@ subroutine rad_cnst_readnl(nlfile)
       call freeunit(unitn)
    end if
 
-   ! whannah - ensure that oldcldoptics=true if using SP 1-mom with RRTMG
```
Comment (brhillman) [inline components/cam/src/physics/cam/rad_constituents.F90]: Note this logic is moved to `build-namelist`.

File: components/cam/src/physics/rrtmg/ebert_curry.F90 [hunk 2]
```diff
@@ -83,114 +81,12 @@ subroutine ec_rad_props_init()
...
-   !call addfld('CIWPTH_NEW','Kg/m2   ',pver, 'I','In Cloud Ice Water Path',phys_decomp, sampling_seq='rad_lwsw')
-   !call addfld('CIWPTH_OLD','Kg/m2   ',pver, 'I','In Cloud Ice Water Path (old)',phys_decomp, sampling_seq='rad_lwsw')
-
    return
 
 end subroutine ec_rad_props_init
 
 !==============================================================================
 
-subroutine cloud_rad_props_get_sw(state, pbuf, &
```
Comment (brhillman) [inline components/cam/src/physics/rrtmg/ebert_curry.F90]: These were not used.

File: components/cam/bld/namelist_files/namelist_definition.xml [hunk 3]
```diff
@@ -4204,12 +4204,6 @@ with modal chemistry.
 Default: none
 </entry>
 
-<entry id="oldcldoptics" type="logical"  category="radiation"
```
Comment (whannah1) [inline components/cam/bld/namelist_files/namelist_definition.xml]: I like the idea of removing this, but is this gonna be a problem for anything on the E3SM side? If not then we might want to just make a parallel PR there to get rid of this option.

File: components/cam/src/physics/rrtmg/radiation.F90 [hunk 4]
```diff
@@ -758,17 +756,20 @@ subroutine radiation_init(phys_state)
...
-            call addfld ('CRM_QRL ', (/'crm_nx_rad','crm_ny_rad','crm_nz    '/), 'I', 'K/s', 'CRM Longwave radiative heating rate' )
-            call addfld ('CRM_QRLC', (/'crm_nx_rad','crm_ny_rad','crm_nz    '/), 'I', 'K/s', 'CRM Longwave radiative heating rate' )
-            call addfld ('CRM_CLD_RAD', (/'crm_nx_rad','crm_ny_rad','crm_nz    '/), 'I', 'fraction', 'CRM cloud fraction' )
-         end if
-       end if
-    end do
+       end if  ! active_calls(icall)
+    end do  ! icall = 0,N_DIAG
+
+    ! Add cloud-scale radiative quantities
```
Comment (brhillman) [inline components/cam/src/physics/rrtmg/radiation.F90]: These were previously inside the `if (active_calls(icall))` check for the rad "diagnostics" calls, but shouldn't have been.

File: components/cam/src/physics/rrtmg/radiation.F90 [hunk 5]
```diff
@@ -1735,49 +1731,47 @@ subroutine radiation_tend( state, ptend,pbuf, &
...
-                end do ! m
-              endif 
             else  ! cldfsnow_idx > 0
               c_cld_tau    (1:nbndsw,1:ncol,:) = cld_tau    (:,1:ncol,:)
               c_cld_tau_w  (1:nbndsw,1:ncol,:) = cld_tau_w  (:,1:ncol,:)
               c_cld_tau_w_g(1:nbndsw,1:ncol,:) = cld_tau_w_g(:,1:ncol,:)
               c_cld_tau_w_f(1:nbndsw,1:ncol,:) = cld_tau_w_f(:,1:ncol,:)
             endif  ! cldfsnow_idx > 0
 
+            ! Save cloud optical depth for CRM column to output later
```
Comment (brhillman) [inline components/cam/src/physics/rrtmg/radiation.F90]: This was previously inside the check on whether or not snow fraction is present, but we want to save the CRM optical properties regardless of whether or not snow is present.

File: components/cam/src/physics/rrtmg/radiation.F90 (hunk 4 above)
Comment (whannah1) [inline components/cam/src/physics/rrtmg/radiation.F90]: Why did you put this outside of the icall loop?

File: components/cam/src/physics/rrtmg/radiation.F90 [hunk 6]
```diff
@@ -1735,49 +1731,47 @@ subroutine radiation_tend( state, ptend,pbuf, &
...
+            if (use_SPCAM) then 
+              do m=1,crm_nz
+                 k = pver-m+1
+                 do i=1,ncol
+                    cld_tau_crm(i,ii,jj,m) =  cld_tau(rrtmg_sw_cloudsim_band,i,k)
+                 end do ! i
+              end do ! m
+            endif 
+
```
Comment (whannah1) [inline components/cam/src/physics/rrtmg/radiation.F90]: Why was this moved?

File: components/cam/src/physics/rrtmg/radiation.F90 (hunk 5 above)
Comment (whannah1) [reply components/cam/src/physics/rrtmg/radiation.F90]: ah, that makes sense, thanks

File: components/cam/src/physics/rrtmg/radiation.F90 (hunk 4 above)
Comment (whannah1) [reply components/cam/src/physics/rrtmg/radiation.F90]: ok, I get it.

File: components/cam/bld/namelist_files/namelist_definition.xml (hunk 3 above)
Comment (brhillman) [reply components/cam/bld/namelist_files/namelist_definition.xml]: It's not used in any of our testing compsets as far as I can tell (according to `grep`), so for all practical purposes it doesn't exist anyways. I'll make another PR upstream.
